src/fetch.rs

```diff
@@ -39,10 +39,17 @@ pub fn get_latest_dependency(
     if env::var("CARGO_IS_TEST").is_ok() {
         // We are in a simulated reality. Nothing is real here.
         // FIXME: Use actual test handling code.
-        return Ok(
-            Dependency::new(crate_name)
-                .set_version(&format!("{}--CURRENT_VERSION_TEST", crate_name)),
-        );
+        if flag_allow_prerelease {
+            return Ok(
+                Dependency::new(crate_name)
+                    .set_version(&format!("{}--PRERELEASE_VERSION_TEST", crate_name)),
```

Comment:
nit: Could you extract `format!("{}--PRERELEASE_VERSION_TEST", crate_name)` and `format!("{}--CURRENT_VERSION_TEST", crate_name)` into one variable `let new_version = if ...` to reduce code repetition?